### PR TITLE
React | Assignment 2  | useEffect and API Data

### PR DESCRIPTION
**Pull request by anandchavan24**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,29 +1,47 @@
...
+  const { data, loading, error } = useFetch('https://jsonplaceholder.typicode.com/todos');
+
+  useEffect(() => {
+    if (data) {
+      setTodos(data);
+    }
+  }, [data]);
+
+
+  const toDoArray:IToDoResp[] | null = data;
```
Comment (Rushikesh-josh) [inline src/App.tsx]: why you are storing API response in constant?

File: src/App.tsx [hunk 2]
```diff
@@ -1,29 +1,47 @@
...
+  
   const [todos, setTodos] = useState(toDoArray);
```
Comment (Rushikesh-josh) [inline src/App.tsx]: remove this you are already setting API value in state inside useEffect

File: src/App.tsx [hunk 3]
```diff
@@ -1,29 +1,47 @@
...
 
-  const markCompleted = (id:number) => {
-    setTodos((prevTodos:ITodo[]) =>
-      prevTodos.map((todo:ITodo) => (todo.id === id ? { ...todo, completed: !todo.completed } : todo))
-    );
+
+  const markCompleted = (id: number) => {
+    if (todos) {
```
Comment (Rushikesh-josh) [inline src/App.tsx:19]: define type of todos here

File: src/App.tsx (hunk 1 above)
Comment (anandchavan24) [reply src/App.tsx]: Resolved

File: src/App.tsx (hunk 2 above)
Comment (anandchavan24) [reply src/App.tsx]: Resolved

File: src/App.tsx (hunk 3 above)
Comment (Rushikesh-josh) [reply src/App.tsx:19]: Please resolve this by adding type

Comment (anandchavan24) [reply src/App.tsx:19]: I have already define type at the TOP

**Review by Rushikesh-josh — APPROVED** [review]

**Pull request merged**